crm-project/src/main/java/CRM/service/TypeService.java

```diff
@@ -2,17 +2,34 @@
 
 import CRM.controller.facades.AttributeFacade;
 import CRM.entity.Attribute;
+import CRM.entity.Status;
+import CRM.entity.Type;
+import CRM.repository.TypeRepository;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TypeService implements AttributeService{
 
     private static Logger logger = LogManager.getLogger(TypeService.class.getName());
 
+    @Autowired
+    private TypeRepository typeRepository;
+
     @Override
     public int update(long id, Attribute object) {
         return 0;
     }
+
+    /**
+     * This function persists a new Type to the database by calling the save function in the TypeRepository class.
+     * @param type The type object to be persisted.
+     * @return The persisted type object.
+     */
+    @Override
+    public Type create(Attribute type) {
+        return typeRepository.save((Type) type);
```

Comment:
Put the function that checks if this attribute already exists in here also.
We can see that it becomes popular as we go, maybe put it in Validations class and make it generic?

Reply:
added, a bit hard to make if generic since they use different repos and strings

Reply:
@Hadi6645 I trust you can do this even if they use different parameters.
I suggest you to look at the method "checkIfIdExists" in the Validations class in util package.
Let me know what you think...

Reply:
it's not same, you use only findById which JpaRepo type recognizes, i use a method that's called by the fields of attribute